### DISCUSSION TITLE
Shelly RGBW 2 Tasmota 13.4.0 no Blue Color

### DISCUSSION THREAD

**Discussion by Meskalito1984** [arendst/Tasmota#20932] [Q&A]
Hello everyone,

I flashed the latest Tasmota version 13.4.0 on both of my Shelly RGBW 2. I use the following template:

{"NAME":"Shelly RGBW2","GPIO":[0,0,288,0.419,1,0,0.416,32,418,417,0,0],"FLAG":0,"BASE":18}

Unfortunately, I now have the problem that I cannot control the blue color on my CCT:RGBWW. If I select blue using the slider on the web interface, my spot lights up green. When I enter the command "color 3" in the console, the spot remains dark. However, I see in the main menu that the spot has been switched on.

Once I was able to control the blue color of the spot with Tasmota, unfortunately this can no longer be reproduced. If I connect the blue wire directly to my power supply, the spot lights up blue.

All other colors (including cold white and warm white) as well as dimming work without any problems. I therefore suspect an error in my GPIO assignment. Can someone help me? I'm starting to feel at a loss.

Thanks in advance!

**Comment by sfromis** [comment]
Why did you use that bad template, instead of just copy/pasting the recommended one from the templates repository. Do not type manually, you have two typos, using `.` where you should have `,`.
Find it here: https://templates.blakadder.com/shelly_RGBW2.html
(Don't worry about it having different numbers, the old standard will be automatically converted)

**Reply by Meskalito1984** [reply]
Thanks for the feedback.  Unfortunately, neither the template from blackadder nor that from https://tasmota.github.io/docs/devices/Shelly-RGBW2/ worked without errors.  I found the template I used in a forum and it ensured that I can see the consumption measurement.

**Reply by sfromis** [reply]
Not sure what you even mean by that. The bad template you posted is the same as the "official" one from Tasmota, except for your typos. You were right about an error in the GPIO assignment, and I pointed out what the error was.

Instead of "neither worked without errors", better be clear about what specific errors you see.

The only difference with the template in the other link you posted now is that the LED is configured as "LedLink" instead of "Led", which is only a minor thing without effect on device operation. Also, that template includes the analog configured for "ADC CT Power", which you can also configure manually. Here's the template updated to include this:
`{"NAME":"Shelly RGBW2","GPIO":[0,0,544,0,419,199,0,0,416,32,418,417,0,4896],"FLAG":0,"BASE":18}`
And this feature was **not** included in the first template you posted.

**Reply by Meskalito1984** [reply]
Thank you very much, but when I use your template, my LED spot flashes constantly and blue doesn't light up either. Here is an excerpt from the log:

_> 23:51:18.295 MQT: stat/tasmota_634F19/POWER = ON
23:51:18.443 MQT: stat/tasmota_634F19/RESULT = {"POWER":"OFF"}
23:51:18.448 MQT: stat/tasmota_634F19/POWER = OFF
23:51:20.506 CMD: color 3
23:51:20.518 MQT: stat/tasmota_634F19/RESULT = {"POWER":"ON","Dimmer":100,"Color":"0000FF00","HSBColor":"240,100,100","White":0,"Channel":[0,0,100,0]}
23:51:21.431 MQT: stat/tasmota_634F19/RESULT = {"POWER":"OFF"}
23:51:21.436 MQT: stat/tasmota_634F19/POWER = OFF
23:51:21.534 MQT: stat/tasmota_634F19/RESULT = {"POWER":"ON"}
23:51:21.538 MQT: stat/tasmota_634F19/POWER = ON
23:51:22.287 MQT: stat/tasmota_634F19/RESULT = {"POWER":"OFF"}
23:51:22.292 MQT: stat/tasmota_634F19/POWER = OFF
23:51:22.420 MQT: stat/tasmota_634F19/RESULT = {"POWER":"ON"}
23:51:22.426 MQT: stat/tasmota_634F19/POWER = ON
23:51:25.287 MQT: stat/tasmota_634F19/RESULT = {"POWER":"OFF"}
23:51:25.294 MQT: stat/tasmota_634F19/POWER = OFF
23:51:25.422 MQT: stat/tasmota_634F19/RESULT = {"POWER":"ON"}
23:51:25.426 MQT: stat/tasmota_634F19/POWER = ON
23:51:26.285 MQT: stat/tasmota_634F19/RESULT = {"POWER":"OFF"}
23:51:26.290 MQT: stat/tasmota_634F19/POWER = OFF
23:51:27.294 MQT: stat/tasmota_634F19/RESULT = {"POWER":"ON"}
23:51:27.299 MQT: stat/tasmota_634F19/POWER = ON
23:51:27.427 MQT: stat/tasmota_634F19/RESULT = {"POWER":"OFF"}
23:51:27.431 MQT: stat/tasmota_634F19/POWER = OFF
23:51:28.294 MQT: stat/tasmota_634F19/RESULT = {"POWER":"ON"}
23:51:28.297 MQT: stat/tasmota_634F19/POWER = ON
23:51:30.257 MQT: stat/tasmota_634F19/RESULT = {"POWER":"OFF"}
23:51:30.261 MQT: stat/tasmota_634F19/POWER = OFF
23:51:30.446 MQT: stat/tasmota_634F19/RESULT = {"POWER":"ON"}
23:51:30.452 MQT: stat/tasmota_634F19/POWER = ON_

 so I used the template from my first post again

**Reply by sfromis** [reply]
Set `Weblog 3` to have details of what happens.

As the template from your first post has major syntax errors as typed, better post the template you are actually using. Command `Backlog Module;Template;Gpio all` provides the details.

**Reply by Meskalito1984** [reply]
With the following template I get the console information below:

Template (I changed GPIO 17 for consumption measurement manually):
{"NAME":"Shelly RGBW2","GPIO":[0,0,288,0.419,1,0,0.416,32,418,417,0,0],"FLAG":0,"BASE":18}


_> 14:25:05.988 CMD: Weblog 3
14:25:05.995 MQT: stat/tasmota_634F19/RESULT = {"WebLog":3}
14:25:06.433 CFG: in Flash gespeichert an FA, Anzahl 130, Bytes 4096
14:25:34.090 CMD: Backlog Module;Template;Gpio all
14:25:34.093 SRC: WebConsole from 192.168.178.75
14:25:34.095 CMD: Grp 0, Cmd 'BACKLOG', Idx 1, Len 24, Pld -99, Data 'Module;Template;Gpio all'
14:25:34.164 SRC: Backlog
14:25:34.168 CMD: Grp 0, Cmd 'MODULE', Idx 1, Len 0, Pld -99, Data ''
14:25:34.173 MQT: stat/tasmota_634F19/RESULT = {"Module":{"0":"Shelly RGBW2"}}
14:25:34.382 SRC: Backlog
14:25:34.385 CMD: Grp 0, Cmd 'TEMPLATE', Idx 1, Len 0, Pld -99, Data ''
14:25:34.394 MQT: stat/tasmota_634F19/RESULT = {"NAME":"Shelly RGBW2","GPIO":[0,0,288,0,419,1,0,0,416,32,418,417,0,4896],"FLAG":0,"BASE":18}
14:25:34.632 SRC: Backlog
14:25:34.635 CMD: Grp 0, Cmd 'GPIO', Idx 1, Len 3, Pld 255, Data 'all'
14:25:34.648 MQT: stat/tasmota_634F19/RESULT = {"GPIO0":{"0":"None"},"GPIO1":{"0":"None"},"GPIO2":{"288":"Led1"},"GPIO3":{"0":"None"},"GPIO4":{"419":"PWM4"},"GPIO5":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"416":"PWM1"},"GPIO13":{"32":"Button1"},"GPIO14":{"418":"PWM3"},"GPIO15":{"417":"PWM2"},"GPIO16":{"0":"None"},"GPIO17":{"4896":"ADC CT Power"}}_
14:26:50.500 UPP: Hue 3 Antwortpakete gesendet zu 192.168.178.44:53889
14:27:44.235 MQT: tele/tasmota_634F19/STATE = {"Time":"2024-03-13T14:27:44","Uptime":"0T00:05:10","UptimeSec":310,"Heap":24,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER":"OFF","Dimmer":100,"Color":"0000FF00","HSBColor":"240,100,100","White":0,"Channel":[0,0,100,0],"Scheme":0,"Fade":"OFF","Speed":1,"LedTable":"ON","Wifi":{"AP":1,"SSId":"HM","BSSId":"B0:F2:08:D4:5A:11","Channel":1,"Mode":"11n","RSSI":72,"Signal":-64,"LinkCount":1,"Downtime":"0T00:00:04"}}
14:27:44.309 MQT: tele/tasmota_634F19/SENSOR = {"Time":"2024-03-13T14:27:44","ANALOG":{"CTEnergy":{"Energy":0.000,"Power":0,"Voltage":230,"Current":0.000}}}
14:28:50.499 UPP: Hue 3 Antwortpakete gesendet zu 192.168.178.44:53889
14:29:03.749 HTP: Hauptmenü
14:29:11.229 SRC: WebGui from 192.168.178.75
14:29:11.235 MQT: stat/tasmota_634F19/RESULT = {"POWER":"ON"}
14:29:11.239 MQT: stat/tasmota_634F19/POWER = ON
14:29:11.413 CFG: in Flash gespeichert an F9, Anzahl 131, Bytes 4096
14:29:17.367 HTP: Konsole

If I use your template with the problem of flashing LED spots, I get the following messages in the console:

_> 14:30:45.325 MQT: tele/tasmota_634F19/SENSOR = {"Time":"2024-03-13T14:30:45","Switch8":"ON","ANALOG":{"CTEnergy":{"Energy":0.000,"Power":5,"Voltage":230,"Current":0.021}}}
> 14:30:45.373 SRC: Switch
> 14:30:45.379 MQT: stat/tasmota_634F19/RESULT = {"POWER":"OFF"}
> 14:30:45.385 MQT: stat/tasmota_634F19/POWER = OFF
> 14:30:45.458 CFG: in Flash gespeichert an F6, Anzahl 134, Bytes 4096
> 14:30:47.335 SRC: WebGui from 192.168.178.75
> 14:30:47.342 MQT: stat/tasmota_634F19/RESULT = {"POWER":"ON"}
> 14:30:47.347 MQT: stat/tasmota_634F19/POWER = ON
> 14:30:47.531 CFG: in Flash gespeichert an F5, Anzahl 135, Bytes 4096
> 14:30:47.538 SRC: Switch
> 14:30:47.544 MQT: stat/tasmota_634F19/RESULT = {"POWER":"OFF"}
> 14:30:47.549 MQT: stat/tasmota_634F19/POWER = OFF
> 14:30:47.638 SRC: Switch
> 14:30:47.645 MQT: stat/tasmota_634F19/RESULT = {"POWER":"ON"}
> 14:30:47.650 MQT: stat/tasmota_634F19/POWER = ON
> 14:30:49.043 SRC: WebGui from 192.168.178.75
> 14:30:49.050 MQT: stat/tasmota_634F19/RESULT = {"POWER":"OFF"}
> 14:30:49.055 MQT: stat/tasmota_634F19/POWER = OFF
> 14:30:49.437 CFG: in Flash gespeichert an F4, Anzahl 136, Bytes 4096
> 14:30:49.904 UPP: Hue 3 Antwortpakete gesendet zu 192.168.178.44:53889
> 14:30:53.567 SRC: WebGui from 192.168.178.75
> 14:30:53.570 CMD: Grp 0, Cmd 'HSBCOLOR', Idx 1, Len 3, Pld 168, Data '168'
> 14:30:53.574 SRC: Light
> 14:30:53.582 MQT: stat/tasmota_634F19/RESULT = {"POWER":"ON","Dimmer":100,"Color":"00FFCC00","HSBColor":"168,100,100","White":0,"Channel":[0,100,80,0]}
> 14:30:54.294 SRC: Switch
> 14:30:54.301 MQT: stat/tasmota_634F19/RESULT = {"POWER":"OFF"}
> 14:30:54.306 MQT: stat/tasmota_634F19/POWER = OFF
> 14:30:54.425 CFG: in Flash gespeichert an FB, Anzahl 137, Bytes 4096
> 14:30:54.431 SRC: Switch
> 14:30:54.438 MQT: stat/tasmota_634F19/RESULT = {"POWER":"ON"}
> 14:30:54.442 MQT: stat/tasmota_634F19/POWER = ON
> 14:30:55.414 CFG: in Flash gespeichert an FA, Anzahl 138, Bytes 4096
> 14:30:55.421 SRC: Switch
> 14:30:55.428 MQT: stat/tasmota_634F19/RESULT = {"POWER":"OFF"}
> 14:30:55.431 MQT: stat/tasmota_634F19/POWER = OFF
> 14:30:56.221 SRC: Switch
> 14:30:56.228 MQT: stat/tasmota_634F19/RESULT = {"POWER":"ON"}
> 14:30:56.233 MQT: stat/tasmota_634F19/POWER = ON
> 14:30:56.324 SRC: Switch
> 14:30:56.335 MQT: stat/tasmota_634F19/RESULT = {"POWER":"OFF"}
> 14:30:56.340 MQT: stat/tasmota_634F19/POWER = OFF
> 14:30:56.426 CFG: in Flash gespeichert an F9, Anzahl 139, Bytes 4096
> 14:30:56.435 SRC: Switch
> 14:30:56.441 MQT: stat/tasmota_634F19/RESULT = {"POWER":"ON"}
> 14:30:56.446 MQT: stat/tasmota_634F19/POWER = ON
> 14:30:57.415 CFG: in Flash gespeichert an F8, Anzahl 140, Bytes 4096
> 14:30:57.478 SRC: WebGui from 192.168.178.75
> 14:30:57.481 CMD: Grp 0, Cmd 'HSBCOLOR', Idx 1, Len 3, Pld 228, Data '228'
> 14:30:57.489 MQT: stat/tasmota_634F19/RESULT = {"POWER":"ON","Dimmer":100,"Color":"0033FF00","HSBColor":"228,100,100","White":0,"Channel":[0,20,100,0]}
> 14:30:58.436 CFG: in Flash gespeichert an F7, Anzahl 141, Bytes 4096
> 14:30:58.446 SRC: Switch
> 14:30:58.454 MQT: stat/tasmota_634F19/RESULT = {"POWER":"OFF"}
> 14:30:58.457 MQT: stat/tasmota_634F19/POWER = OFF
> 14:30:59.258 SRC: Switch
> 14:30:59.267 MQT: stat/tasmota_634F19/RESULT = {"POWER":"ON"}
> 14:30:59.271 MQT: stat/tasmota_634F19/POWER = ON
> 14:31:00.109 SRC: Switch
> 14:31:00.116 MQT: stat/tasmota_634F19/RESULT = {"POWER":"OFF"}
> 14:31:00.122 MQT: stat/tasmota_634F19/POWER = OFF
> 14:31:00.282 SRC: Switch
> 14:31:00.292 MQT: stat/tasmota_634F19/RESULT = {"POWER":"ON"}
> 14:31:00.297 MQT: stat/tasmota_634F19/POWER = ON
> 14:31:01.542 HTP: Konsole
> 14:31:01.644 SRC: Switch
> 14:31:01.652 MQT: stat/tasmota_634F19/RESULT = {"POWER":"OFF"}
> 14:31:01.657 MQT: stat/tasmota_634F19/POWER = OFF
> 14:31:02.261 SRC: Switch
> 14:31:02.268 MQT: stat/tasmota_634F19/RESULT = {"POWER":"ON"}
> 14:31:02.272 MQT: stat/tasmota_634F19/POWER = ON
> 14:31:08.274 SRC: Switch
> 14:31:08.281 MQT: stat/tasmota_634F19/RESULT = {"POWER":"OFF"}
> 14:31:08.286 MQT: stat/tasmota_634F19/POWER = OFF
> 14:31:08.392 CFG: in Flash gespeichert an F6, Anzahl 142, Bytes 4096
> 14:31:10.269 SRC: Switch
> 14:31:10.275 MQT: stat/tasmota_634F19/RESULT = {"POWER":"ON"}
> 14:31:10.280 MQT: stat/tasmota_634F19/POWER = ON
> 14:31:10.401 CFG: in Flash gespeichert an F5, Anzahl 143, Bytes 4096
> 14:31:10.408 SRC: Switch
> 14:31:10.414 MQT: stat/tasmota_634F19/RESULT = {"POWER":"OFF"}
> 14:31:10.421 MQT: stat/tasmota_634F19/POWER = OFF_

**Reply by sfromis** [reply]
Not sure what happened with the template I posted, but it is indeed broken. Oops. Here's what I meant to post:
`{"NAME":"Shelly RGBW2","GPIO":[0,0,288,0,419,1,0,0,416,32,418,417,0,4896],"FLAG":0,"BASE":18}`

Except for your typos, which I pointed out, this is the same as your original template when adding `ADC CT Power`

The way Tasmota handles such typos is not to complain, but silently drop the bad values, resulting in a template like this
`{"NAME":"Shelly RGBW2","GPIO":[0,0,288,0,1,0,0,32,418,417,0,0,0,4896],"FLAG":0,"BASE":18}` instead of
`{"NAME":"Shelly RGBW2","GPIO":[0,0,288,0.419,1,0,0.416,32,418,417,0,4896],"FLAG":0,"BASE":18}`
The PWM components are moved around and two of them lost, which will cause a mess with colors.

However, what you posted was **not** the template you used, as the saved template was:
`{"NAME":"Shelly RGBW2","GPIO":[0,0,288,0,419,1,0,0,416,32,418,417,0,4896],"FLAG":0,"BASE":18}` 

This is exactly the same as the one I meant to post.

Now back to the original question about colors not working. Are you still seeing that with the correct template? (meaning without the typos)

**Reply by Meskalito1984** [reply]
Thank you very much for the explanation, I was actually not aware of Tasmota's procedure.


> Now back to the original question about colors not working. Are you still seeing that with the correct template? (meaning without the typos)

yes, unfortunately the spot remains dark when you enter “color 3” on the console.  If I select the color blue using the console slider, the spot lights up light green.  It doesn't matter which template I use, the result is always identical.

**Reply by sfromis** [reply]
Mixing up of blue/green sounds like bad wiring.

I'd suggest a test with `MultiPWM 1` to give you 4 sliders/toggles on the front page, one for each of red, green, blue and white, respectively. By trying the toggles and sliders, you can make sure that the wiring for each of the 4 channels is correct, and fix as indicated. Of course, you can go back to normal mode with `MultiPWM 0`.

**Reply by Meskalito1984** [reply]
thanks for the tip. Here I have the same effect, when I press the third control (I guess it's blue) the spot stays dark.

To be honest, I rule out incorrect wiring because the spot lights up blue when connected directly to the ground of the power supply. I have the same equipment twice (2x the power supply, 2x the Shelly RGBW2, 16 x the LED spots) and the problem occurs in both configurations.

I made a request in the Shelly Forum because I wanted to rule out a new hardware revision of the RGBW2, unfortunately this research was unsuccessful, which is why I suspected Tasmota. Unfortunately I didn't try the original Shelly FW with the rest of the equipment but instead flashed Tasmota right at the beginning.

I am now absolutely at a loss.

**Reply by sfromis** [reply]
Well, did you verify all 4 toggles/sliders for colors red, green, blue, white respectively? If yes, does red/green/white work as they should? If only blue does nothing, that indicates either wrong gpio (which could be a new board revision), or bad wiring. I'm asking because you seemed to indicate that blue/green was somehow mixed up, and just want to be clear about what exactly you see.

The above would fit with you never seeing the blue channel do anything. If that's the case, you could try moving PWM3 to other gpios, and see if you can find the blue on another one.

**Reply by Meskalito1984** [reply]
All other sliders worked correctly and the corresponding colors lit up and dimming the colors worked without any problems.  

I was once able to regulate the blue color, unfortunately I cannot reproduce this and no longer remember which template I used.  

I will check different GPIO configurations again.  could it possibly be due to the optocoupler used for the 230v ac circuit?  If I understood it correctly, in this case the GPIO 5 should be on the switch, which I haven't consciously taken into account yet.

**Reply by sfromis** [reply]
I'd not expect the usage of the [AC Switch](https://tasmota.github.io/docs/devices/Shelly-RGBW2/#using-the-attached-ac-switch-adapter) function to have any relation to individual PWM channels working or not.

But still not understanding how this related to you seeing red, green and white working.
> If I select the color blue using the console slider, the spot lights up light green.

Or maybe you used a slider in a way not *only* producing blue, but also green? Such things was why I suggested MultiPWM to be absolutely clear about what channels you are messing with.

Anyway, I do not have the device, and thus no specific experience with any oddities.
